### PR TITLE
Added VIPGRP functions matching *-AddressGroup*

### PR DESCRIPTION
**Pull request by poundy**

### CODE REVIEW

**Comment by alagoutte** [comment]
Thanks !, it will make a quick review,

do you have try all cmdlet ?

File: PowerFGT/Private/Confirm.ps1 [hunk 1]
```diff
@@ -4,6 +4,8 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 #
+# PR from Poundy - Add Confirm-FGTVipGroup
```
[the remark this answers is not in the document]
Comment (poundy) [reply PowerFGT/Private/Confirm.ps1]: do you mean my comment, or the function ?

**Comment by poundy** [comment]
> Thanks !, it will make a quick review,
> 
> do you have try all cmdlet ?

Yes, I've tested the full cmdlet suite. Not extensively, but I've done tests on everything on my FW setup.

I've made amendments based on changes requested, but I am not familiar with whether these changes have now been re-submitted into this for approval again or if my "request review" has just been done on the older commit?  I saw it re-do the automated checks so I assume at some level those updates have made it here. (sorry for being such a newbie on github and contributions)

**Comment by alagoutte** [comment]
Thanks Brett ! it will make some test (i think no before next week)
but look (very) good !

You can also update README.(md) for add example ;-)

There is no yet unit test (but it is on my roadmap for next release...)

File: README.md [hunk 2]
```diff
@@ -264,7 +266,7 @@ Actually, support only `equal` and `contains` filter type
 ### Address Group
 
 You can create a new Address Group `Add-FGTFirewallAddressGroup`, retrieve its information `Get-FGTFirewallAddressGroup`,
-modify its properties `Set-FGTFirewallAddressGroup`, copy/clone its properties `Copt-FGTFirewallAddressGroup`,
+modify its properties `Set-FGTFirewallAddressGroup`, copy/clone its properties `Copy-FGTFirewallAddressGroup`,
```
Comment (alagoutte) [inline README.md]: Good catch ! thanks

File: README.md [hunk 3]
```diff
@@ -8,6 +8,8 @@ With this module (version 0.4.1) you can manage:
 
 - [Address](#Address) (Add/Get/Copy/Set/Remove object type ipmask/subnet)
 - [AddressGroup](#Address-Group) (Add/Get/Copy/Set/Remove and Add/Remove Member)
+- Vip (Add/Get/Copy/Set/Remove VIP)
```
Comment (alagoutte) [inline README.md]: There is already a Virtual IP chapiter (and it is sort by alpha)

**Comment by alagoutte** [comment]
Thanks, i will try this week !

File: PowerFGT/Public/cmdb/firewall/vipgroup.ps1 [hunk 4]
```diff
@@ -0,0 +1,592 @@
...
+
+        Add VIP Group with member MyVip1 and a comment, associated to interface wan1
+    #>
+
+    Param(
+        [Parameter (Mandatory = $true)]
+        [string]$name,
+        [Parameter (Mandatory = $true)]
+        [string[]]$member,
+        [Parameter (Mandatory = $true)]
```
Comment (alagoutte) [inline PowerFGT/Public/cmdb/firewall/vipgroup.ps1:39]: interface is really needed ?

File: PowerFGT/Public/cmdb/firewall/vipgroup.ps1 [hunk 5]
```diff
@@ -0,0 +1,592 @@
...
+        #Add member to Member Array
+        $members = @( )
+        foreach ( $m in $member ) {
+            $member_name = @{ }
+            $member_name.add( 'name', $m)
+            $members += $member_name
+        }
+        $vipgrp | add-member -name "member" -membertype NoteProperty -Value $members
+
+        if ( $PsBoundParameters.ContainsKey('interface') ) {
```
Comment (alagoutte) [inline PowerFGT/Public/cmdb/firewall/vipgroup.ps1:81]: if interface is really needed, you need to remove this if

File: PowerFGT/Public/cmdb/firewall/vipgroup.ps1 (hunk 4 above)
Comment (poundy) [reply PowerFGT/Public/cmdb/firewall/vipgroup.ps1:39]: can you check the API docs to confirm it's shown as mandatory there? I am pretty sure that you can't define it without a bound interface via  GUI/CLI... just checked and confirmed that Interface is required: 
config firewall vipgrp
    edit "Brett_Test"
(Brett_Test) # next
Attribute 'interface' MUST be set.
Command fail. Return code 1

File: PowerFGT/Public/cmdb/firewall/vipgroup.ps1 (hunk 5 above)
Comment (poundy) [reply PowerFGT/Public/cmdb/firewall/vipgroup.ps1:81]: OK, but functionally it works, because there always will be an interface, this tests true always.

Comment (poundy) [reply PowerFGT/Public/cmdb/firewall/vipgroup.ps1:81]: (so a more correct statement would be "if interface is really needed, you should remove this redundant if")

Comment (poundy) [reply PowerFGT/Public/cmdb/firewall/vipgroup.ps1]: amendment complete

File: PowerFGT/Public/cmdb/firewall/vipgroup.ps1 (hunk 4 above)
Comment (alagoutte) [reply PowerFGT/Public/cmdb/firewall/vipgroup.ps1:39]: it is no specified on API :-/ 
after, we can use the same idea like for VIP cmdlet (and also GUI), set any by default  https://github.com/FortiPower/PowerFGT/blob/master/PowerFGT/Public/cmdb/firewall/vip.ps1#L49

**Comment by alagoutte** [comment]
Hi @poundy can you look if it is ok for you ? (i have some test and fix some issue found with when write test...)

@CedricMoreau, can look review too ?

**Comment by poundy** [comment]
Just commenting in here because I am kind of stuck at the moment, not really understanding the push-pull status here on what I can use to test.
I am happy to re-load a cut of the code just can't be sure where I should be grabbing it from? If you could help guide me there, I will get it into my env and test against a 6.2.x device.

**Comment by alagoutte** [comment]
Hi @poundy, i prepare to merge soon this patch (and why i have fix a typo and rebase change!)

**Comment by poundy** [comment]
> Hi @poundy, i prepare to merge soon this patch (and why i have fix a typo and rebase change!)

I am happy to load again and add to my (production) tests if that helps; I have been using it in the original form I originally committed although vipgrps aren't something I create a lot of!

**Review by CedricMoreau — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -21,6 +21,7 @@ With this module (version 0.4.1) you can manage:
 - System Global (Get)
 - [VDOM](#VDOM) (Get)
 - [Virtual IP](#Virtual-IP) (Get/Add/Remove object type static-nat)
+- VirtualIPGroup (Add/Get/Copy/Set/Remove and Add/Remove Member)
```
Comment (CedricMoreau) [inline README.md:24]: Use the same format as the one above

**Comment by alagoutte** [comment]
Thanks @poundy and sorry for delay ;-)

**Pull request merged**